test/environmentSettings.ts

```diff
@@ -55,10 +57,13 @@ export class EnvironmentSettings {
         process.env.VSCODE_VERSION ||
         EnvironmentSettings._instance._vscodeVersion;
 
-      if (process.env.SPEC_FILES) {
-        EnvironmentSettings._instance._specFiles = [
-          './test/specs/**/' + process.env.SPEC_FILES
-        ];
+      if (process.env.SPEC_FILES?.length) {
+        EnvironmentSettings._instance._specFiles = process.env.SPEC_FILES.split(
+          ','
+        ).map(specFile => `./test/specs/**/${specFile.trim()}`);
+        utilities.log(
```

Comment:
`getInstance()` is starting to get a bit long.  I'm thinking maybe the initialization code should be moved into the constructor:

```

  private constructor() {
    this._vscodeVersion =
      process.env.VSCODE_VERSION ||
      EnvironmentSettings._instance._vscodeVersion;

    if (process.env.SPEC_FILES?.length) {
      this._specFiles = process.env.SPEC_FILES.split(',').map(
        specFile => `./test/specs/**/${specFile.trim()}`
      );
      utilities.log(`SPEC_FILES: ${this._specFiles.join(', ')}`);
    }

    this._devHubAliasName =
      process.env.DEV_HUB_ALIAS_NAME || this._devHubAliasName;

    this._devHubUserName =
      process.env.DEV_HUB_USER_NAME || this._devHubUserName;

    this._extensionPath = process.env.EXTENSION_PATH || this._extensionPath;

    this._throttleFactor =
      parseInt(process.env.THROTTLE_FACTOR!) || this._throttleFactor;

    if (process.env.WITH_TEST_PROJECT) {
      this._sfdxProjectHome = join(
        __dirname,
        '..',
        '..',
        process.env.WITH_TEST_PROJECT
      );
    }
  }

  public static getInstance(): EnvironmentSettings {
    if (!EnvironmentSettings._instance) {
      EnvironmentSettings._instance = new EnvironmentSettings();
    }

    return EnvironmentSettings._instance;
  }
```

thoughts?

Reply:
I'll move it up into ctor